tedana/utils.py

```diff
@@ -198,7 +199,12 @@ def dice(arr1, arr2, axis=None):
         dsi = np.zeros(arr_sum.shape)
     else:
         intersection = np.logical_and(arr1, arr2)
-        dsi = (2. * intersection.sum(axis=axis)) / arr_sum
+        if np.any(arr_sum == 0):
+            with warnings.catch_warnings():
+                warnings.simplefilter("ignore")
```

Comment:
```suggestion
                warnings.simplefilter("ignore", RuntimeWarning)
```

If we're specific enough about the warning we catch, we could just get rid of the entire if statement starting at line 198. Right now we check if _all_ thresholded maps are empty, then here we check if some are empty, and in the new `else` we run when none are empty. We should be able to just filter the warning and calculate across all maps without checking if any are empty, right?

Reply:
Or I guess we could mask things and unmask them? Maybe with a debug message reporting the number of empty maps?

Reply:
I agree with your suggestion, I actually meant to add that filter, whoops.
Yes, we can just not check for emptiness, but I'm inclined to think we actually _should_ warn users if _all_ of the maps are empty. The issue is it's difficult to make a meaningful warning in that context.

Reply:
What about a warning of how many are empty? Then we don't have to check if all are empty, then any are empty. We check if any are empty, raise a warning with how many, and then outside that if statement we apply the warning filter?

Reply:
That would be pretty sensible and straightforward. We also can add a prompt for the users to examine the component table for dice values of 0.

Reply:
I like it!